src/components/application_manager/src/event_engine/event_dispatcher_impl.cc

```diff
@@ -72,7 +72,8 @@ void EventDispatcherImpl::raise_event(const Event& event) {
 
 void EventDispatcherImpl::add_observer(const Event::EventID& event_id,
                                        int32_t hmi_correlation_id,
-                                       EventObserver* observer) {
+                                       EventObserver* const observer) {
+  DCHECK(observer);
```

Comment:
@Kozoriz I don't think we have to use DCHECK here. We just need to avert adding of NULL pointer objects. Here should be used `if `statement instead of assert.